geonode/geoserver/signals.py

```diff
@@ -139,6 +139,11 @@ def geoserver_pre_save(instance, sender, **kwargs):
         if getattr(ogc_server_settings, "BACKEND_WRITE_ENABLED", True):
             gs_catalog.save(gs_layer)
 
+    if gs_resource:
+        if gs_resource.keywords:
+            for keyword in gs_resource.keywords:
+                instance.keywords.add(keyword)
```

Comment:
@afabiani I think I can add a related keyword it the layer is already existing.
I agree with you that this can be problematic when creating a new layer. Need to figure out what is the best way to handle it